web/src/main/java/org/cbioportal/web/util/StudyViewFilterUtil.java

```diff
@@ -153,4 +154,117 @@ public Integer getFilteredCountByDataEquality(List<ClinicalDataFilter> attribute
+        Map<String, String> patientIdToStudyId = mapCaseToStudy(patientIds, studyIdsOfPatients);
+        Map<String, Boolean> sampleAttributeIdLookup = listToMap(sampleAttributeIds);
+        Map<String, Boolean> patientAttributeIdLookup = listToMap(patientAttributeIds);
+        Map<String, Boolean> conflictingPatientAttributeIdLookup = listToMap(conflictingPatientAttributes);
+            
+        if (CollectionUtils.isNotEmpty(sampleAttributeIds)) {
+            combinedResult.addAll(
+                filterClinicalDataByStudyAndSampleAndAttribute(
+                    unfilteredClinicalDataForSamples,
+                    sampleIdToStudyId,
+                    sampleAttributeIdLookup
+                )
+            );
+        }
+
+        if (CollectionUtils.isNotEmpty(patientAttributeIds)) {
+            combinedResult.addAll(
+                filterClinicalDataByStudyAndPatientAndAttribute(
+                    unfilteredClinicalDataForPatients,
+                    patientIdToStudyId,
+                    patientAttributeIdLookup
+                )
+            );
+        }
+
+        if (CollectionUtils.isNotEmpty(conflictingPatientAttributes)) {
+            combinedResult.addAll(
+                filterClinicalDataByStudyAndPatientAndAttribute(
+                    unfilteredClinicalDataForConflictingPatientAttributes,
+                    patientIdToStudyId,
+                    conflictingPatientAttributeIdLookup
+                )
+            );
+        }
+
+        return combinedResult;
+    }
+    
+    private List<ClinicalData> filterClinicalDataByStudyAndSampleAndAttribute(
+        List<ClinicalData> clinicalData,
+        Map<String, String> sampleToStudyId,
+        Map<String, Boolean> attributeIdLookup
+    ) {
+        return clinicalData
+            .stream()
+            .filter(d ->
+                sampleToStudyId.getOrDefault(d.getSampleId(), "").equals(d.getStudyId()) &&
+                attributeIdLookup.getOrDefault(d.getAttrId(), false)
+            )
+            .collect(Collectors.toList());
+    }
+
+    private List<ClinicalData> filterClinicalDataByStudyAndPatientAndAttribute(
+        List<ClinicalData> clinicalData,
+        Map<String, String> patientToStudyId,
+        Map<String, Boolean> attributeIdLookup
+    ) {
+        return clinicalData
+            .stream()
+            .filter(d ->
+                patientToStudyId.getOrDefault(d.getPatientId(), "").equals(d.getStudyId()) &&
+                attributeIdLookup.getOrDefault(d.getAttrId(), false)
+            )
+            .collect(Collectors.toList());
+    }
+    
+    private <T> Map<T, Boolean> listToMap(List<T> list) {
+        return list.stream().collect(Collectors.toMap(s -> s, s -> true, (s1, s2) -> s1));
+    }
+
+    private  Map<String, String> mapCaseToStudy(List<String> caseIds, List<String> studyIds) {
+        Map<String, String> caseToStudy = new HashMap<>();
+        
+        for (int i =0; i < caseIds.size(); i++) {
```

Comment:
there is a possibility for duplicate caseId. combination of case id and study id should be the key

Reply:
Good catch! A single patient/sample can be found in more than one study, right?

Reply:
yes